modules/ravestate_roboyqa/__init__.py

```diff
@@ -0,0 +1,144 @@
+
+    category = None
+    memory_info = None
+
+    # question word: What?
+    if triple.is_question(QuestionWord.OBJECT):
+        if triple.match_either_lemma(pred={"like"}, subj={"hobby"}):
+            category = "HAS_HOBBY"
+        elif triple.match_either_lemma(pred={"learn"}, subj={"skill"}):
+            category = "skills"
+        elif triple.match_either_lemma(pred={"can"}, subj={"ability"}):
+            category = "abilities"
+        elif triple.match_either_lemma(subj={"age"}):
+            category = "age"
+            memory_info = roboy_age(roboy.get_properties(key="birthdate"))
+        elif triple.match_either_lemma(subj={"name"}):
+            category = "full_name"
+            memory_info = roboy.get_properties(key=category)
+        elif triple.match_either_lemma(pred={"become"}):
+            category = "future"
+    # question word: Where?
+    elif triple.is_question(QuestionWord.PLACE):
+        if triple.match_either_lemma(pred={"be"}):
+            category = "FROM"
+        elif triple.match_either_lemma(pred={"live"}):
+            category = "LIVE_IN"
+    # question word: Who?
+    elif triple.is_question(QuestionWord.PERSON):
+        if triple.match_either_lemma(obj={"father", "dad"}):
+            category = "CHILD_OF"
+        elif triple.match_either_lemma(obj={"brother", "sibling"}):
+            category = "SIBLING_OF"
+        elif triple.match_either_lemma(obj={"friend", "girlfriend"}):
+            category = "FRIEND_OF"
+        else:
+            category = "full_name"
+            memory_info = roboy.get_properties(key=category)
+    elif triple.match_either_lemma(obj={"part", "member"}):
+        category = "MEMBER_OF"
+    # question word: How?
+    elif triple.is_question(QuestionWord.FORM):
+        if triple.match_either_lemma(pred={"old"}):
+            category = "age"
+            memory_info = roboy_age(roboy.get_properties(key="birthdate"))
+        elif triple.match_either_lemma(pred={"be"}):
+            category = "well_being"
+    elif triple.match_either_lemma(obj={"skill"}):
+        category = "skills"
+    elif triple.match_either_lemma(obj={"ability"}):
+        category = "abilities"
+
+    if category and category.isupper() and not isinstance(roboy.get_relationships(key=category), dict):
+        node_id = random.sample(roboy.get_relationships(key=category),1)[0]
+        memory_info = sess.retrieve(node_id=int(node_id))[0].get_name()
+
+    elif category and category.islower() and not isinstance(roboy.get_properties(key=category), dict):
+        property_list = [x.strip() for x in roboy.get_properties(key=category).split(',')]
+        memory_info = random.sample(property_list, 1)[0]
+
+    if memory_info:
+        ctx["rawio:out"] = verbaliser.get_random_successful_answer(category) % memory_info
+    elif category == "well_being":
+        ctx["rawio:out"] = verbaliser.get_random_successful_answer(category)
+    else:
+        ctx["rawio:out"] = "Sorry I do not know."
```

Comment:
this method is getting really big - 100 lines, would split it in multiple methods to make it easier to understand and test 😄

Reply:
maybe even split it up into multiple states? but unfortunately we're a bit out of time on this one. will create an issue.

Reply:
See #37